src/main/java/io/kensu/collector/config/DamProcessEnvironment.java

```diff
@@ -33,78 +22,45 @@ public String damIngestionToken(){
+        Map<String, String> m = new HashMap<>();
+        m.put("KENSU_PROCESS_NAME", this.getProcessName());
+        m.put("KENSU_LAUCHED_BY_USER", this.getProcessLauncherUserRef());
+        m.put("KENSU_RUN_ENVIRONMENT", this.getRunEnvironment());
+        m.put("KENSU_PROJECTS", this.getDamProjectRefs());
+        m.put("KENSU_CODEBASE_LOCATION", getOptEnv("DAM_CODEBASE_LOCATION", ""));
+        m.put("KENSU_CODE_VERSION", getOptEnv("DAM_CODE_VERSION", ""));
+        return m;
     }
 
-    public CodeVersionRef getExecutedCodeVersionRef(DamBatchBuilder batchBuilder) {
-        String codeBaseLocation = getOptEnv("DAM_CODEBASE_LOCATION", null);
-        String codeVersionValue = getOptEnv("DAM_CODE_VERSION", null);
-        if (codeBaseLocation != null && codeVersionValue != null) {
-            CodeBase codeBase = new CodeBase().pk(new CodeBasePK().location(codeBaseLocation));
-            CodeVersionPK codeVersionPK = new CodeVersionPK()
-                    .codebaseRef(new CodeBaseRef().byPK(codeBase.getPk()))
-                    .version(codeVersionValue);
-            CodeVersion codeVersion = new CodeVersion().pk(codeVersionPK);
-            batchBuilder.add(codeBase);
-            batchBuilder.add(codeVersion);
-            return new CodeVersionRef().byPK(codeVersionPK);
-        }
-        return null;
+    public String getRunEnvironment(){
+        return getOptEnv("DAM_RUN_ENVIRONMENT", "");
     }
 
-    public ProcessRun enqueProcessRun(Process process,
-                                         String endpointName,
-                                         DamBatchBuilder batchBuilder) {
-        String processName = process.getPk().getQualifiedName();
-        ProcessRun processRun =  new ProcessRun()
-                .pk(new ProcessRunPK()
-                        .qualifiedName(String.format("%s :: %s run at %d", processName, endpointName, System.currentTimeMillis()))
-                        .processRef(new ProcessRef().byPK(process.getPk())))
-                .launchedByUserRef(getProcessLauncherUserRef(batchBuilder))
-                .environment(getRunEnvironment())
-                .projectsRefs(getDamProjectRefs(batchBuilder))
-                .executedCodeVersionRef(getExecutedCodeVersionRef(batchBuilder));
-        batchBuilder.add(processRun);
-        return processRun;
+    public String getDamProjectRefs(){
+        String projectsStr = getOptEnv("DAM_PROJECTS", "");
+        return projectsStr;
+//        if (projectsStr != null){
+//            List<String> projects = Arrays.stream(projectsStr.trim()
+//                                            .split(";")).map(String::trim)
+//                                            .filter(e->e.length()>0)
+//                                            .collect(Collectors.toList());
+//            if (projects.size() == 0) return null;
+//            return projects;
+//        } else {
+//            return null;
+//        }
```

Comment:
comment?

Reply:
good catch. indeed here the idea was to further think if it's a good idea to send string like"project1;project2" without parsing the string (e.g. json would be a more standard format), but keep things simple for simplicity it's probably fine, java is quite nasty language, and there are even worse ones...